app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -201,6 +205,49 @@ open class HomeActivity : AppCompatActivity(), ShareFragment.TabsSharedCallback
         openToBrowser(BrowserDirection.FromGlobal, customTabSessionId)
     }
 
+    @SuppressWarnings("ComplexMethod")
+    private fun handleDeepLink(uri: Uri) {
+        val link = uri.host
+
+        // Handle links that require more than just simple navigation
+        when (link) {
+            "enable_private_browsing" -> {
+                navHost.navController.navigate(NavGraphDirections.actionGlobalHomeFragment())
+                browsingModeManager.mode = BrowsingMode.Private
+            }
+            "make_default_browser" -> {
+                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) { return }
+                val settingsIntent = Intent(
+                    android.provider.Settings.ACTION_MANAGE_DEFAULT_APPS_SETTINGS
+                )
+                startActivity(settingsIntent)
+            }
+            "open" -> {
+                uri.getQueryParameter("url")?.let {
+                    load(
+                        searchTermOrURL = it,
```

Comment:
Should we do any type of sanitization on this url? cc: @colintheshots

Reply:
So we already allow VIEW intents with potentially malicious site URLs or data URIs. The only change here is the schema is only recognized by our app currently and may load without asking first which app to use. However, there is no rule that another browser cannot also register fenix: and be annoying. It might be nice to use Android App Links to avoid contention over deep links with other apps. https://developer.android.com/training/app-links

Reply:
👍 Created a follow up issue here: https://github.com/mozilla-mobile/fenix/issues/4738